liquidctl/cli.py

```diff
@@ -191,6 +213,29 @@ def _list_devices(devices, using_filters=False, device_id=None, verbose=False, d
     assert 'device' not in opts or len(devices) <= 1, 'too many results listed with --device'
 
 
+def _dev_status_obj(dev, status):
+    # don't suppress devices without status data (typically from initialize)
+    if not status:
+        status = []
+
+    return {
+        'bus': dev.bus,
+        'address': dev.address,
+        # suppress the experimental suffix (check .experimental from list)
+        'description': dev.description.replace(' (experimental)', ''),
+        # JSON values can only be objects, arrays, numbers, strings, or
+        # false/null/true; convert all non-numeric status values to strings
```

Comment:
This is too extreme, or at least shouldn't be relied on by consumers of this JSON.

In the future, arrays, tuples and other types that can be represented in JSON should probably be kept.  Although, IIRC, we don't currently have any of those; the closest are time spans, which could perhaps be serialized as spans of [milli]seconds, instead of a ISO-like time string.

I need to think about this a bit more... @MarshallAsch, what do  you think?

Reply:
Personally I think having the timespans as milliseconds would be a lot better than  using [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals) time intervals because those look nasty imo. And if interval strings were used then they would have be parsed and generated and thats no fun.